### PR TITLE
feat(frontend/backend/shared): add my tasks screen bb-335

### PR DESCRIPTION
**Pull request by marharita08**
## Summary
Added 'My tasks' screen, where user current tasks is shown.

### Frontend:
/tasks - 'My tasks' screen
### Backend:
GET: /tasks/current - getting current tasks of current user.

## Screenshot

image

### CODE REVIEW

File: apps/backend/src/modules/tasks/task.repository.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+	create(): Promise<unknown> {
+		return Promise.resolve();
+	}
+
+	public delete(): ReturnType<Repository["delete"]> {
+		return Promise.resolve(true);
+	}
+
+	find(): Promise<unknown> {
```
Comment (luiqor) [inline apps/backend/src/modules/tasks/task.repository.ts:23]: For both task service and task repository:
B30. All class properties and methods are marked with member access (private , public or protected).

File: apps/backend/src/db/migrations/20240910113839_add_tasks_table.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+const TaskStatus = {
+	COMPLETED: "Completed",
+	CURRENT: "Current",
+	SKIPPED: "Skipped",
+} as const;
+
+const DELETE_STRATEGY = "CASCADE";
+
+async function up(knex: Knex): Promise<void> {
+	await knex.schema.createTable(TableName.TASKS, (table) => {
```
Comment (luiqor) [inline apps/backend/src/db/migrations/20240910113839_add_tasks_table.ts:30]: When adding a migration with a new table, you should also add the table to the diagram in README.md.

File: packages/shared/src/modules/tasks/libs/types/task-dto.type.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { type ValueOf } from "src/libs/types/types.js";
```
Comment (fshabanov) [inline packages/shared/src/modules/tasks/libs/types/task-dto.type.ts:1]:
```suggestion
import { type ValueOf } from "../../../../libs/types/types.js";
```

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/frontend/src/assets/css/variables.css [hunk 4]
```diff
@@ -17,8 +18,8 @@
 	--physical-end: #fff8b5;
 	--work-start: #69ff35;
 	--work-end: #c3ff19;
-	--friend-start: #7f21ce;
-	--friend-end: #cb00ff;
+	--friends-start: #7f21ce;
```
Comment (ihorLysak) [inline apps/frontend/src/assets/css/variables.css:21]: This might break gradients for others

Comment (luiqor) [reply apps/frontend/src/assets/css/variables.css:21]: I agree. I used these colors for sliders. With updated name, then we will need to update these colors across the app.

**Review by luiqor — APPROVED** [review]

File: apps/backend/src/modules/tasks/task.repository.ts [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+				})
+			: null;
+	}
+
+	public async findAll(): Promise<TaskEntity[]> {
+		const tasks = await this.taskModel
+			.query()
+			.withGraphFetched(`[${RelationName.CATEGORY}]`);
+
+		return tasks.map((task) =>
```
Comment (fshabanov) [inline apps/backend/src/modules/tasks/task.repository.ts:79]: B27

File: apps/backend/src/modules/tasks/task.repository.ts [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+		);
+	}
+
+	public async findCurrentByUserId(userId: number): Promise<TaskEntity[]> {
+		const tasks = await this.taskModel
+			.query()
+			.withGraphFetched(`[${RelationName.CATEGORY}]`)
+			.where({ status: TaskStatus.CURRENT, userId });
+
+		return tasks.map((task) =>
```
Comment (fshabanov) [inline apps/backend/src/modules/tasks/task.repository.ts:101]: B27

File: apps/backend/src/modules/tasks/task.service.ts [hunk 7]
```diff
@@ -0,0 +1,62 @@
+import { type Service } from "~/libs/types/types.js";
+
+import { type TaskDto } from "./libs/types/types.js";
+import { type UsersTaskCreateDto } from "./libs/types/users-task-create-dto.js";
```
Comment (fshabanov) [inline apps/backend/src/modules/tasks/task.service.ts:4]:
```suggestion
import { type UsersTaskCreateDto } from "./libs/types/types.js";
```

File: apps/backend/src/modules/tasks/libs/types/users-task-create-dto.ts [hunk 8]
```diff
@@ -0,0 +1,5 @@
+import { type TaskCreateDto } from "./types.js";
```
Comment (fshabanov) [inline apps/backend/src/modules/tasks/libs/types/users-task-create-dto.ts:1]: Change file name to `users-task-create-dto.type.ts`

File: apps/frontend/src/pages/tasks/libs/constants/constants.ts [hunk 9]
```diff
@@ -0,0 +1,3 @@
+const ZERO_ARRAY_LENGTH = 0;
```
Comment (fshabanov) [inline apps/frontend/src/pages/tasks/libs/constants/constants.ts:1]: Use `NumericValue` enum

File: apps/frontend/src/pages/tasks/tasks.tsx [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+} from "~/libs/hooks/hooks.js";
+import { actions as taskActions } from "~/modules/tasks/tasks.js";
+
+import { TaskCard } from "./libs/components/components.js";
+import { ZERO_ARRAY_LENGTH } from "./libs/constants/constants.js";
+import styles from "./styles.module.css";
+
+const Tasks: React.FC = () => {
+	const dispatch = useAppDispatch();
+	const { dataStatus, tasks } = useAppSelector(({ tasks }) => ({
```
Comment (fshabanov) [inline apps/frontend/src/pages/tasks/tasks.tsx:16]: B27

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/frontend/src/pages/tasks/libs/helpers/convert-category-name-to-css-class-name/convert-category-name-to-css-class-name.helper.ts [hunk 11]
```diff
@@ -0,0 +1,5 @@
+const convertCategyNameToCssClassName = (categoryName: string): string => {
```
Comment (luiqor) [inline apps/frontend/src/pages/tasks/libs/helpers/convert-category-name-to-css-class-name/convert-category-name-to-css-class-name.helper.ts:1]:
```suggestion
const convertCategoryNameToCssClassName = (categoryName: string): string => {
```

**Review by luiqor — APPROVED** [review]

**Review by WilhelmWesser — APPROVED** [review]

File: apps/backend/src/modules/tasks/libs/types/types.ts [hunk 12]
```diff
@@ -0,0 +1,2 @@
+export { type UsersTaskCreateDto } from "./users-task-create-dto.type.js";
```
Comment (fshabanov) [inline apps/backend/src/modules/tasks/libs/types/types.ts:1]:
```suggestion
export { type UsersTaskCreateRequestDto } from "./users-task-create-request-dto.type.js";
```

File: apps/frontend/src/pages/tasks/libs/helpers/convert-category-name-to-css-class-name/convert-category-name-to-css-class-name.helper.ts [hunk 13]
```diff
@@ -0,0 +1,5 @@
+const convertCategoryNameToCssClassName = (categoryName: string): string => {
```
Comment (fshabanov) [inline apps/frontend/src/pages/tasks/libs/helpers/convert-category-name-to-css-class-name/convert-category-name-to-css-class-name.helper.ts:1]:
```suggestion
const convertCategoryNameToKebabCase = (categoryName: string): string => {
```
Let's not limit to css classNames

File: apps/frontend/src/pages/tasks/tasks.tsx [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+	const isTasksEmpty = tasks.length === NumericalValue.ZERO;
+
+	return (
+		<>
+			<h4 className={styles["title"]}>My Tasks</h4>
+			<div className={styles["board"]}>
+				{isLoading ? (
+					<Loader />
+				) : (
+					!isTasksEmpty &&
```
Comment (fshabanov) [inline apps/frontend/src/pages/tasks/tasks.tsx:37]:
```suggestion
```
This is unnecessary

**Review by fshabanov — CHANGES_REQUESTED** [review]

**Review by fshabanov — APPROVED** [review]

**Pull request merged**